generateZonefile.py

```diff
@@ -110,5 +91,57 @@ def main():
     f.close()
 
 
+def generate_node_hostname(node):
+    node_hostname_regex = "^[0-9a-zäöü]([0-9a-zäöü-]{0,38}[0-9a-zäöü])?$"
+    node_replace_regex = "([^0-9a-zäöü])+"
+
+    if not re.match(node_hostname_regex, node["hostnameLower"]):
+        new_hostname = re.sub(
+            node_replace_regex,
+            '-',
+            node["hostnameLower"]
+        ).strip('-')
+        warning("replaced: \t" + node["hostnameLower"] + " with " + new_hostname)
+        node["hostnameLower"] = new_hostname
+
+    if not re.match(node_hostname_regex, node["hostnameLower"]):
```

Comment:
Why that if? It seems similar to the one in 98 :thinking:

Reply:
The first is before the replacement to just run the replacement regex if it would be "useful" (ignoring the `--` problem), the second is to catch errors that the replace can noit solve like to long host names

Reply:
ah, thanks

Reply:
Fixed by replacing all hostnames